annotation-processor/src/test/java/org/hibernate/validator/ap/ConstraintValidationProcessorTest.java

```diff
@@ -96,10 +100,48 @@ public void fieldLevelValidationUsingBuiltInConstraints() {
 		);
 	}
 
-	/**
-	 * HV-567
-	 */
+
+	@Test
+	@TestForIssue( jiraKey = "HV-840" )
+	public void overridingMethodParameterConstraintsTest() {
+		File sourceFile = compilerHelper.getSourceFile( MethodOverridingTests.class );
+
+		boolean compilationResult = compilerHelper.compile( new ConstraintValidationProcessor(), diagnostics, sourceFile );
+
+		assertFalse( compilationResult );
+		assertThatDiagnosticsMatch(
+				diagnostics,
+				new DiagnosticExpectation( Kind.ERROR, 39 ),
+				new DiagnosticExpectation( Kind.ERROR, 56 ),
+				new DiagnosticExpectation( Kind.ERROR, 71 ),
+				new DiagnosticExpectation( Kind.ERROR, 90 ),
+				new DiagnosticExpectation( Kind.ERROR, 144 ),
+				new DiagnosticExpectation( Kind.ERROR, 152 ),
+				new DiagnosticExpectation( Kind.ERROR, 169 ),
+				new DiagnosticExpectation( Kind.ERROR, 191 ),
+				new DiagnosticExpectation( Kind.ERROR, 219 ),
+				new DiagnosticExpectation( Kind.ERROR, 373 ),
+				new DiagnosticExpectation( Kind.ERROR, 387 ),
+				new DiagnosticExpectation( Kind.ERROR, 409 ),
+				new DiagnosticExpectation( Kind.ERROR, 434 )
+		);
+
+		assertEquals( diagnostics.getDiagnostics().get( 0 ).getMessage( Locale.getDefault() ),
+				"Parameters of method \"doSomething((@javax.validation.constraints.NotNull :: java.lang.String))\" do not respect the inheritance rules. " +
```

Comment:
The error will be shown in the context of the method. Do we need to repeat the entire signature? Shouldn't the compiler itself point it out?